### ISSUE TITLE
How to decide module's learnable parameters at initialisation

### ISSUE THREAD

**Issue by itk22** [patrick-kidger/equinox#841]
Hi Patrick, 
Is it possible to "freeze" specific module parameters at initialization based on a condition? In the example below, I want to make the `scale` parameter non-learnable if and only if `modulate_scale` is set to `False`.

```
import equinox as eqx
import jax
import jax.numpy as jnp
import numpy as np


class FiLM(eqx.Module):
    scale: jax.Array | np.ndarray
    shift: jax.Array | np.ndarray

    def __init__(self, in_features, modulate_scale, modulate_shift):
        self.scale = (
            jnp.ones((in_features,))
            if modulate_scale
            else np.ones((in_features,))
        )

        self.shift = (
            jnp.zeros((in_features,))
            if modulate_shift
            else np.zeros((in_features,))
        )

    def __call__(self, x):
        return self.scale * x + self.shift


film = FiLM(10, True, True)
print(eqx.partition(film, eqx.is_array))
```
I've realized that `eqx.partition` tags both JAX arrays and NumPy arrays as learnable. This behavior wasn't what I initially expected, as I thought it would only tag JAX arrays. Given this observation, I am also curious to know if the NumPy arrays are converted under the hood for compatibility with JAX transforms.

**Comment by patrick-kidger** [comment]
Probably the simplest way to avoid gradients would be to use `jax.lax.stop_gradient` during `__call__` in your desired case.

NumPy arrays and JAX arrays have to be treated interchangeably, unfortunately. JAX's own internals sometimes substitue JAX arrays for NumPy arrays, so if we do otherwise then some of JAX's own debug tooling will explode :)

**Comment by itk22** [comment]
That's clear, thanks Patrick!
